daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -87,6 +87,7 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
     */
   def submit(
       submitters: Set[Party],
+      readAs: Set[Party],
```

Comment:
Can you extend the code documentation above, please?

Reply:
Good idea, thanks!

Reply:
Thanks extending the code documentation!
If these are really additional readers (i.e., besides the submitters), I'd suggest naming the parameter accordingly. Though, I might be missing some context regarding why we chose `readAs` originally.

Reply:
I’d prefer using the same name everywhere. Using different names for the same thing across our codebase is just confusing. As for why we have those names, I believe we tried to match the names we used for authorization when we added multitparty submissions.

Reply:
The `readAs` and `actAs` terms were first used when we added ledger API authorization, for parties the bearer of the auth token is allowed to use for read-only (e.g., transaction service) and state modifying (e.g., command service) ledger API queries, respectively. At this time, the concept of multi-party submissions did not exist and `readAs` had nothing to do with contract visibility during command interpretation.

The terms then made it into the ledger API and the engine when we added multi-party submissions, where `readAs` means "parties that can be used to look up contracts/keys, in addition to the submitters". Using a `readAs` party for multi-party submissions requires a corresponding `readAs` claim in the auth token.

Note also the following details:
- The `readAs`/ `actAs` parties from auth tokens are a superset of the parties used in ledger API commands (you don't have to use all parties you are allowed to)
- The `readAs`/ `actAs` parties from ledger API commands are a superset of the minimal sets of parties required to successfully submit the command (you can submit a command as Alice+Bob, even though Alice alone would be enough to authorize the command in the Daml model)
- The ledger API server normalizes `readAs` parties by removing `actAs` parties (as `actAs` implies read access)

`submitters` and `actAs` are the probably the same thing.